.github/workflows/release_mac.yml

```diff
@@ -0,0 +1,64 @@
+name: MacRelease
+
+on:
+  push:
+    branches:
+      - rel-*
+  # test this PR for now, remove it before the merge
+  pull_request:
+    branches: [ master ]
+env:
+  MACOSX_DEPLOYMENT_TARGET: 10.9
+jobs:
+  build:
+    runs-on: macos-10.15
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        architecture: ['x64']
+    steps:
+    - uses: actions/checkout@v2
+    - name: Checkout submodules
+      shell: bash
+      run: |
+        auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+        git submodule sync --recursive
+        git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: ${{ matrix.architecture }}
+        miniconda-version: latest
+    - name: Install python dependencies
+      env:
+        ONNX_BUILD_TESTS: 1
+        USE_MSVC_STATIC_RUNTIME: 1
+        ONNX_ML: 1
+        CMAKE_ARGS: "-DONNX_USE_LITE_PROTO=ON -DONNXIFI_DUMMY_BACKEND=ON -DONNX_WERROR=ON"
+      run: |
+        conda create -qy -n py${{ matrix.python-version }} python=${{ matrix.python-version }} -c anaconda
+        conda init bash
+        source ~/.bash_profile
+        conda activate py${{ matrix.python-version }}
```

Comment:
https://github.com/actions/setup-python/issues/26
Try miniconda to obtain Python with lower MACOSX_DEPLOYMENT_TARGET. Current Python agent in GitHub Action is using Python with 10.14 (MACOSX_DEPLOYMENT_TARGET) and it will produce 10.14 wheel anyway.